pyiron_contrib/atomistic/atomicrex/potential_factory.py

```diff
@@ -0,0 +1,145 @@
+import posixpath
+import xml.etree.ElementTree as ET
+
+from pyiron_base import PyironFactory, InputList
+
+from pyiron_contrib.atomistic.atomicrex.function_factory import FunctionFactory
+from pyiron_contrib.atomistic.atomicrex.utility_functions import write_pretty_xml
+
+class ARPotFactory(PyironFactory):
+    @staticmethod
+    def eam_potential(identifier="EAM", export_file="output.eam.fs", rho_range_factor=2.0, resolution=10000, species=["*","*"]):
+        return EAMPotential(
+            identifier=identifier,
+            export_file=export_file,
+            rho_range_factor=rho_range_factor,
+            resolution=resolution,
+            species=species,
+        )
+
+    @staticmethod
+    def lennard_jones_potential(sigma, epsilon, cutoff, species={"a": "*", "b": "*"}, identifier="LJ"):
+        return LJPotential(sigma, epsilon, cutoff, species=species, identifier=identifier)
+
+
+class AbstractPotential(InputList):
```

Comment:
@pmrv Here I subclass inputlist and then the other potentials are subclassed from this class.